tests/ui/conftest.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+"""Plugin for running UI tests."""
+
+import socket
+
+import pytest
+
+from redash import (
+    create_app,
+    redis_connection,
+)
+from redash.models import db
+
+from tests.factories import Factory
+from tests.ui.live_server import LiveServer
+from tests.ui.pages.login import LoginPage
+
+
+@pytest.fixture(name='host')
+def fixture_host():
+    """Return a host name as str."""
+    return 'localhost'
+
+
+@pytest.fixture(name='port')
+def fixture_port():
+    """Return a free port as int."""
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    port = s.getsockname()[1]
+
+    # Avoid 'address already in use' errors
+    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    s.close()
+    return port
+
+
+@pytest.fixture(name='app')
+def fixture_app(host, port):
```

Comment:
😍 for not falling for the fixture naming trap

Reply:
https://twitter.com/hackebrot/status/981600024332259328 😁